minHeap/minHeap.go

```diff
@@ -0,0 +1,95 @@
+package heap
+
+import (
+	"errors"
+	"fmt"
+)
+
+type heap struct {
+	store    []int
+	capacity int
+	size     int
+}
+
+//creates a heap from an int array
+func buildMinHeap(values []int) (*heap, error) {
+	if len(values) == 0 {
+		return nil, errors.New("Array is empty")
+	}
+	minHeap := heap{capacity: (len(values) * 2), size: 0}
```

Comment:
why capacity default to twice size only ?
what if i wanna build a heap size of 1 right now and keep adding new vaues as we go?

Reply:
I was thinking of table-doubling based on the need for capacity to make efficient use of the memory

Reply:
good idea